packages/agents/watcher/src/operations/validateAndPause.ts

```diff
@@ -1,31 +1,33 @@
 import { PauseResponse, ReportEventType } from "@connext/nxtp-adapters-watcher";
-import { createLoggingContext, createMethodContext, RequestContext } from "@connext/nxtp-utils";
+import { createMethodContext, RequestContext } from "@connext/nxtp-utils";
 
 import { getContext } from "../watcher";
 
-export const validateAndPause = async () => {
+export const validateAndPause = async (requestContext: RequestContext) => {
   const {
     adapters: { watcher },
   } = getContext();
 
-  const { requestContext } = createLoggingContext("validateAndPause");
-
-  const { needsPause, reason } = await watcher.checkInvariants(requestContext);
+  const { needsPause, reason, transactions } = await watcher.checkInvariants(requestContext);
   if (needsPause) {
-    await pauseAndAlert(requestContext, reason || "");
+    await pauseAndAlert(requestContext, reason || "", transactions);
   }
 };
 
-export const pauseAndAlert = async (requestContext: RequestContext, reason: string): Promise<PauseResponse[]> => {
+export const pauseAndAlert = async (
+  requestContext: RequestContext,
+  reason: string,
+  transactions?: Record<string, string[]>,
+): Promise<PauseResponse[]> => {
   const {
     adapters: { watcher },
     logger,
     config,
   } = getContext();
-  const methodContext = createMethodContext("pauseAndAlert");
+  const methodContext = createMethodContext(pauseAndAlert.name);
 
   const domains = Object.keys(config.chains);
-  logger.warn("Pausing contracts!!!", requestContext, methodContext, { reason });
+  logger.warn("Pausing contracts!!!", requestContext, methodContext, { reason, transactions, domains });
```

Comment:
Good first step here; I think ideally we should have some way to view these transactions in the alert itself, but obviously it's going to be a really long list.

Reply:
Would be ideal to have a table of txs we can view somewhere that shows amounts being transferred / delivered of diff tokens (even more ideally, ordered by amount).

What we want to provide here is a list of "suspects": at some point, we violated the invariant because some subset of these transactions are minting more than should be possible.

Technically, you can present these transactions chronologically and 1 of the txs will be the one that pushes us over the maximum amount we'd expect to be bridged/minted, BUT that tx isn't necessarily the perpetrator. It could be that the attacker pushed us right UP to the limit with fraudulent mints, and a subsequent tx is what actually did it. It could also be a series of txs that does this.

There's no telling just from amounts alone which txs are fraudulent... or is there? Important to ask: why do we want to know which fraudulent tx did the dirty? We can track down EOA, etc. but really it doesn't change the fact that we should be pausing.

Basically, I'm realizing that the critical info here we want to know is which of the txs' mints have a corresponding deposit on the sending chain!

If we check every tx's `transferId` and then check on the tx's `origin` whether an appropriate `xcall` exists that involves a deposit for the expected amount, we can quickly deduce whether there actually ARE fraudulent txs and which txs are fraudulent and which are not!

Reply:
cleared after each interval

Reply:
created an issue to track separately: https://github.com/connext/monorepo/issues/3196